File: source/windowsMobile/SubProtocol/Ping/ControlI/PingHandlerI.cs
/***************************************************************************
----------------------------------------------------------------------------
  This file is part of the HLMP API - Plug-ins Source Code.
  http://hlmprotocol.bicubic.cl
 
  Copyright (C) Bicubic TMG.  All rights reserved.
 
  This source code is intended only as a supplement to HLMP API 
  and/or on-line documentation.  
 
  THE SOURCE CODE CONTAINED WITHIN THIS FILE AND ALL RELATED      
  FILES OR ANY PORTION OF ITS CONTENTS SHALL AT NO TIME BE        
  COPIED, TRANSFERRED, SOLD, DISTRIBUTED, OR OTHERWISE MADE       
  AVAILABLE TO OTHER INDIVIDUALS WITHOUT EXPRESS WRITTEN CONSENT  
  AND PERMISSION FROM BICUBIC TMG.   

  THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
  KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
  IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
  PARTICULAR PURPOSE.
----------------------------------------------------------------------------
****************************************************************************/

using System;
using System.Collections.Generic;
using System.Text;
using CommLayerCompact;
using CommLayerCompact.Messages;

namespace SubProtocolCompact.Ping.ControlI
{
    /// <summary>
    /// Interfaz para eventos de mensajes de ping
    /// </summary>
    public interface PingHandlerI
    {        
        /// <summary>
        /// Se gatilla cuando se ha recibido la confirmacion de un ping
        /// </summary>
        /// <param name="netUser">El autor de la respuesta de ping</param>
        /// <param name="milliseconds">Milisegundos totales</param>
        void pingResponseMessageReceived(NetUser netUser, Double milliseconds);
    }
}

File: source/java/hlmp/SubProtocol/Ping/ControlI/PingHandlerI.java
package hlmp.SubProtocol.Ping.ControlI;

import hlmp.CommLayer.NetUser;

/**
 * Interfaz para eventos de mensajes de ping
 * @author nicolas
 *
 */
public interface PingHandlerI {

	/**
	 * Se gatilla cuando se ha recibido la confirmacion de un ping
	 * @param netUser El autor de la respuesta de ping
	 * @param milliseconds Milisegundos totales
	 */
	public void pingResponseMessageReceived(NetUser netUser, long milliseconds);
}
